### PR TITLE
Issue #13301: Write a test to ensure checks are present in menu

### LINKED ISSUE #13301: Write a test to ensure checks are present in menu
From https://github.com/checkstyle/checkstyle/pull/13295#pullrequestreview-1496521604

> Why are some checks present in the sidebar, but not in the RHS check type summary?

> Example: https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/0f81099_2023154343/checks/annotation/index.html is missing MissingOverride. Is there some way to write a test for this?

---
A test needs to be written to ensure that the checks listed in the sidebar/menu are the same as the in the table in category `index.xml` page.

For example, Annotations lists these items in the menu
[AnnotationLocation](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/0f81099_2023154343/checks/annotation/annotationlocation.html)
    [AnnotationOnSameLine](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/0f81099_2023154343/checks/annotation/annotationonsameline.html)
    [AnnotationUseStyle](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/0f81099_2023154343/checks/annotation/annotationusestyle.html)
    [MissingDeprecated](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/0f81099_2023154343/checks/annotation/missingdeprecated.html)
    [MissingOverride](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/0f81099_2023154343/checks/annotation/missingoverride.html)
    [PackageAnnotation](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/0f81099_2023154343/checks/annotation/packageannotation.html)
    [SuppressWarnings](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/0f81099_2023154343/checks/annotation/suppresswarnings.html)
    [SuppressWarningsHolder](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/0f81099_2023154343/checks/annotation/suppresswarningsholder.html)

But only those are in the table at `checks/annotation/index.xml`

AnnotationLocation | Checks location of annotation on language elements.
-- | --
AnnotationOnSameLine | Checks that annotations are located on the same line with their targets.
AnnotationUseStyle | Checks the style of elements in annotations.
MissingDeprecated | Verifies that the annotation @Deprecated and the Javadoc tag               @deprecated               are both present when either of them is present.
PackageAnnotation | Checks that all package annotations are in the package-info.java file.
SuppressWarnings | Allows to specify what warnings that               @SuppressWarnings               is not allowed to suppress.

`MissingOverride` is missing. I am fixing it in #13295 but a test needs to be written.

### PR DESCRIPTION
**Pull request by stoyanK7**
Resolves #13301 

Adds a test to ensure that every module xdoc has a link in the menu.

### CODE REVIEW

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Please show failure in CI, just add a commit with missing link that we created issue on

**Comment by stoyanK7** [comment]
https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=15158&view=logs&j=eb841e9f-6e1e-5c29-3ad5-74ec2c658069&t=a9b761e6-b8d5-50cf-467a-54967189cab2&l=1219
```
[ERROR] Failures: 
[ERROR]   XdocsPagesTest.testAllConfigsHaveLinkInSite:258 Expected to find link to 'href="checks/annotation/missingoverride.html"' in src/site/site.xml
expected to contain:
    href="checks/annotation/missingoverride.html"
but was:
    <?xml version="1.0" encoding="UTF-8"?>
....

```

**Review by nrmancuso — APPROVED** [review]
I think the message that truth provides is sort of weird `but was: <?xml version="1.0" encoding="UTF-8"?>`, but I am not sure of an easy way to change this.

**Comment by romani** [comment]
@stoyanK7 , windows test execution is failing, please fix

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java [hunk 1]
```diff
@@ -243,6 +244,21 @@ private static boolean isPresent(String availableChecks, String checkName) {
         return availableChecks.matches(linkPattern);
     }
 
+    @Test
+    public void testAllConfigsHaveLinkInSite() throws Exception {
+        final String siteContent = Files.readString(SITE_PATH);
+
+        for (Path path : XdocUtil.getXdocsConfigFilePaths(XdocUtil.getXdocsFilePaths())) {
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:251]: I am bit confused.
getXdocsConfigFilePaths find all files by prefix config
https://github.com/checkstyle/checkstyle/blob/c3ab3fd9eb4dddfaba54921a192a0be9f0a06f44/src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocUtil.java#L79
 
But we do not have such files after split https://github.com/checkstyle/checkstyle/tree/master/src/xdocs

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java (hunk 1 above)
Comment (stoyanK7) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:251]: We have an issue for that. I have to do it:

https://github.com/checkstyle/checkstyle/issues/13132

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:251]: Ok, but what is reason of this PR ? 
Should we combine two issues?

Comment (stoyanK7) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:251]: This PR came from  https://github.com/checkstyle/checkstyle/pull/13295#pullrequestreview-1496521604

I will do #13132 in another PR

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:251]: Ok, please confirm that this PR is not doing any valuable validation.

Ad if so, can we close it and simply wait for 13123.

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:251]: If there is some benefit please share.

Comment (nrmancuso) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:251]: Example of failure is here: https://github.com/checkstyle/checkstyle/pull/13327#issuecomment-1613267207

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:251]: According to code it should not do violation, as Check is not file "config_". 
So I am confused.

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:251]: ok, I misread this getXdocsConfigFilePaths method.
it works
image

**Review by romani — APPROVED** [review]
ok to merge

**Comment by romani** [comment]
Ci is restrated.

ok to merge if CI s greeen

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java [hunk 2]
```diff
@@ -243,6 +244,22 @@ private static boolean isPresent(String availableChecks, String checkName) {
...
+            final String expectedFile = path.toString()
+                    .replace(".xml", ".html")
+                    .replaceAll("\\\\", "/")
```
Comment (nrmancuso) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:254]: Added this for windows file path compatibility

**Comment by romani** [comment]
One failure:
```
0a1,22
> <td><i><a class="externalLink" href="https://www.oracle.com/java/technologies/faq-sun-packages.html">https://www.oracle.com/java/technologies/faq-sun-packages.html</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/java/technologies/faq-sun-packages.html">https://www.oracle.com/java/technologies/faq-sun-packages.html</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/java/technologies/java-ee-glance.html">https://www.oracle.com/java/technologies/java-ee-glance.html</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/java/technologies/javamail.html">https://www.oracle.com/java/technologies/javamail.html</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/java/technologies/javase/codeconventions-contents.html">https://www.oracle.com/java/technologies/javase/codeconventions-contents.html</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence">https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence">https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence">https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence">https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence">https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence">https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence">https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence">https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence">https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence">https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence">https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence">https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#firstsentence</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html">https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/ma14-architect-annotations.html">https://www.oracle.com/technical-resources/articles/java/ma14-architect-annotations.html</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/java/ma14-architect-annotations.html">https://www.oracle.com/technical-resources/articles/java/ma14-architect-annotations.html</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/javase/finalization.html">https://www.oracle.com/technical-resources/articles/javase/finalization.html</a>: 403 Forbidden</i></td></tr>
> <td><i><a class="externalLink" href="https://www.oracle.com/technical-resources/articles/javase/finalization.html">https://www.oracle.com/technical-resources/articles/javase/finalization.html</a>: 403 Forbidden</i></td></tr>
------------ g
```

**Pull request merged**